### DISCUSSION TITLE
Help: build install Transmission on openwrt system

### DISCUSSION THREAD

**Discussion by Judgegao** [transmission/transmission#6651] [Q&A]
### What is the issue?

I'm a newbie. I want to compile and install transmission on the openwrt system (hardware Xiaomi router).
My router device information is:
- Hostname: ImmortalWrt
- Host model: Xiaomi Redmi Router AX6000
- Architecture: ARMv8 Processor rev 4 (v8l) x 4 (1998MHz)
- Target platform: mediatek/mt7986
- Firmware version: ImmortalWrt 18.06-5.4-SNAPSHOT r11814-ef0c86bdb0 / LuCI branch (git-22.323.17670-f9380b5)
- Kernel version: 5.4.224

Openwrt officially provides a binary package installation method (https://openwrt.org/docs/guide-user/services/downloading_and_filesharing/transmission), but does not provide a tutorial for compilation and installation. Because I made some modifications based on transmission(version 4.0.4), I need to compile and install it. 

The official compilation and installation documentation was not found (https://github.com/transmission/transmission/blob/main/docs/Building-Transmission.md). I hope the experts can tell me how to do it! help me!

### Which application of Transmission?

None

**Comment by killemov** [comment]
@Judgegao wrote:
> I'm a newbie. I want to compile and install transmission on the openwrt system (hardware Xiaomi router).

Then this will be a steep learning curve, but keep at it!

The key point is that the "Building Transmission" guide does not include building for "other" architectures. You are probably working from an Intel/AMD computer and the target is an ARM router. This implies that not only the machine instructions are different, but also the ecosystem, such as any libraries you might need to (hard)link.

Luckily the GCC can compile for other architectures, see: https://gcc.gnu.org/onlinedocs/gcc/AArch64-Options.html

This might be a good place to start: https://openwrt.org/user/ja/my-openwrt-firmware-image-build-guide

At least this should get you to compile code for you platform.

**Comment by tearfur** [comment]
I've never messed with openwrt, but generally package repositories have their build scripts available open-source, you can look at how they build their packages there.

For openwrt, https://github.com/openwrt/packages/tree/openwrt-21.02/net/transmission seems to be their Transmission package's build scripts.

**Comment by killemov** [comment]
What are the rules for issue <-> discussion transformation? I'd say the ISSUE is a lack in documentation.

**Reply by tearfur** [reply]
I'd say the author of this post just wanted to look for help building on his system. So let's keep it that way for him.

If you think this post shows a lack of documentation, you can make a new issue referencing this post.

**Comment by rezad1393** [comment]
If I remember correctly you needed a sdk from openwrt for your hardware.
then you create a build folder for it or better yet download the one from openwrt itself from here for main repo 
https://github.com/openwrt/packages/tree/master/net/transmission

but yours seems to be a fork of openwrt for china so maybe this?
https://github.com/immortalwrt/packages/tree/openwrt-18.06/net/transmission

and then use this guide:
https://openwrt.org/docs/guide-developer/toolchain/single.package

by the way why are you on this build?

openwrt seems to support your device officially.
https://openwrt.org/toh/xiaomi/redmi_ax6000
https://downloads.openwrt.org/releases/23.05.3/targets/mediatek/filogic/

though it is a bit complicated and if you are a noob you may brick your device installing it.
though you have a fork on openwrt installed already so you are not on stock so you may have a way easier install than from stock to openwrt.
